src/api/mod.rs

```diff
@@ -23,8 +23,37 @@ pub mod api_client;
 pub mod error;
 pub mod rpc_api;
 
+use ac_node_api::EventDetails;
 use serde::{Deserialize, Serialize};
 
+/// Extrinsic report returned upon a submit_and_watch request.
+/// Holds as much information as available.
+#[derive(Debug, Clone)]
+pub struct ExtrinsicReport<Hash> {
```

Comment:
We should definitely add decode and encode here, maybe also serialize and deserialize. :)

Reply:
Agreed, I'll add it for now!

Reply:
Ok, fine with me.

Reply:
https://github.com/scs/substrate-api-client/issues/397